ToDoList/Repository/WeatherRepository.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  WeatherRepository.swift
+//  ToDoList
+//
+//  Created by Nathi Mabena on 2025/08/20.
+//
+
+import Foundation
+import CoreLocation
+
+protocol WeatherRepositoryProtocol {
+    func fetchCurrentWeather(lat: Double, lon: Double) async throws -> Weather
+    func fetchForecast(lat: Double, lon: Double) async throws -> [Forecast]
+}
+
+final class WeatherRepository: WeatherRepositoryProtocol {
+    
+    private let baseURL = "https://api.weatherapi.com/v1"
```

Comment:
```suggestion
    private let baseURLString = "https://api.weatherapi.com/v1"
```
This is better for readability as it explains that its a string and not the actual URL